modules/apps/commerce/commerce-service/src/main/resources/META-INF/custom-sql/default.xml

```diff
@@ -154,6 +154,16 @@
 				CommerceOrder.modifieddate DESC
 		]]>
 	</sql>
+	<sql id="com.liferay.commerce.service.persistence.CommerceOrderFinder.findFirstCommerceOrderDate">
+		<![CDATA[
+			SELECT
+				MIN(CommerceOrder.orderDate) AS FIRST_DATE
```

Comment:
@luca-pellizzon This query is expensire, as is without filters and context, even companies

Reply:
we should definitely filter by companyId. but as a note is it not possible to set the min date for the filter to minium possible date (1/1/1970) and then preselect today's date - 1 month as fromValue in the filter? @luca-pellizzon @FabioDiegoMastrorilli